tools/elf2efi.lds

```diff
@@ -6,6 +6,11 @@ SECTIONS {
         .text ALIGN(CONSTANT(MAXPAGESIZE)) : {
                 *(.text .text.*)
         }
+
+        /* When linking a minimal addon stub the linker can merge .text and .dynsym creating a RWE segment,
+         * and then rejects it. Ensure there's a gap. */
+        . = ALIGN (CONSTANT (COMMONPAGESIZE));
```

Comment:
Please use `MAXPAGESIZE` to keep it consistent. (And use the same spacing style as the rest of the linker script)